server/app/services/scheduler/filter/affinity.rb

```diff
@@ -87,70 +76,32 @@ def split_affinity(affinity)
 
       # @param [String] comparator
       # @return [Boolean]
-      def soft?(comparator)
-        comparator.end_with?('~')
+      def soft?(flags)
+        flags && flags.include?('~')
       end
 
       # @param [HostNode] node
-      # @param [String] compare
       # @param [String] value
-      def node_match?(node, compare, value)
-        if compare == '=='
-          node.name == value
-        elsif compare == '!='
-          node.name != value
-        else
-          false
-        end
+      def node_match?(node, value)
+        node.name == value
       end
 
       # @param [HostNode] node
-      # @param [String] compare
       # @param [String] value
-      def container_match?(node, compare, value)
-        container_names = node.containers.map{|c|
-          c.labels['io;kontena;container;name'].to_s
-        }
-        if compare == '=='
-          container_names.any?{|n| n == value}
-        elsif compare == '!='
-          !container_names.any?{|n| n == value}
-        end
+      def container_match?(node, value)
+        node.containers.where(name: value).exists?
```

Comment:
Simplified the `labels['io;kontena;container;name']` to just query by `name` for convenience, but that may not be correct... I'm not sure what the significance of matching by the label is, because AFAICT the two should always be the same for service containers?

The only difference is that matching by label prevents using affinities like `container==weave`, but is that intentional?

Reply:
Simple is better.